### PR TITLE
Tokens Page: Inverse Token Previews

### PR DESCRIPTION
**Pull request by zahnster**
* Feature Branch Merge to website/new-tokens-page
* Adds support for inverse background colors on Token Previews for inverse tokens
* Adds test and stories

### CODE REVIEW

File: packages/paste-website/src/components/tokens-list/index.tsx [hunk 1]
```diff
@@ -71,8 +71,11 @@ export const TokensList: React.FC<TokensListProps> = (props) => {
   }
 
   const backgroundColor =
-    (tokens['background-colors'].find((token) => token.name === 'color-background-body')?.value as string) ??
-    'rgb(255,255,255)';
+    (tokens['background-colors'].find((token) => token.name === 'color-background-body')?.value as string) ?? '#fff';
+
+  const inverseBackgroundColor =
+    (tokens['background-colors'].find((token) => token.name === 'color-background-body-inverse')?.value as string) ??
+    '#121C2D';
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/index.tsx:78]: chore: can you add some comments above these explaining what they're for? I'm having a hard time grokking this code at a glance so a comment can help future maintainers

File: packages/paste-website/src/components/tokens-list/token-card/index.tsx [hunk 2]
```diff
@@ -75,9 +75,12 @@ export const TokenCard: React.FC<{
   name: Token['name'];
   value: Token['value'];
   backgroundColor: Properties['backgroundColor'];
+  inverseBackgroundColor: Properties['backgroundColor'];
   comment?: Token['comment'];
   useCamelCase?: boolean;
-}> = ({category, name, value, comment, backgroundColor, useCamelCase}) => {
+  ['data-testid']?: string;
+}> = (props) => {
+  const {category, name, value, comment, backgroundColor, inverseBackgroundColor, useCamelCase} = props;
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/token-card/index.tsx:83]:
```suggestion
}> = ({category, name, value, comment, backgroundColor, inverseBackgroundColor, useCamelCase, ...props}) => {
```

File: packages/paste-website/src/components/tokens-list/token-card/index.tsx [hunk 3]
```diff
@@ -120,8 +123,15 @@ export const TokenCard: React.FC<{
       minHeight="sizeSquare170"
       marginBottom="space40"
       overflow="hidden"
+      data-testid={props['data-testid']}
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/token-card/index.tsx:126]: thought: might be fine to spread props here

File: packages/paste-website/src/components/tokens-list/token-card/token-example/index.tsx [hunk 4]
```diff
@@ -15,9 +15,12 @@ type TokenExampleProps = {
   name: string;
   value: string;
   backgroundColor: Properties['backgroundColor'];
+  inverseBackgroundColor: Properties['backgroundColor'];
 };
 
-export const TokenExample: React.FC<TokenExampleProps> = ({category, name, value, backgroundColor}) => {
+export const TokenExample: React.FC<TokenExampleProps> = (props) => {
+  const {category, name, value, inverseBackgroundColor} = props;
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/token-card/token-example/index.tsx:22]:
```suggestion
export const TokenExample: React.FC<TokenExampleProps> = ({category, name, value, inverseBackgroundColor, ...props}) => {
```

File: packages/paste-website/src/components/tokens-list/token-card/token-example/index.tsx [hunk 5]
```diff
@@ -15,9 +15,12 @@ type TokenExampleProps = {
...
+  const backgroundColor = name.toLowerCase().match('inverse') ? inverseBackgroundColor : props.backgroundColor;
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/token-card/token-example/index.tsx:23]: praise: nice change!

**Review by TheSisb — APPROVED** [review]

**Comment by katieporter** [comment]
issue: the contrast check score on inverse text tokens blends into the background—can that swap to use color-text-inverse?
image

**Comment by zahnster** [comment]
Good catch, @katieporter - I've made some adjustments:

Screen Shot 2022-07-19 at 10 04 06 AM

I also added a filter to not display deprecated tokens - this appears to be in line with the current Tokens Page. It was a slight regression as we went from consuming gatsby data to consuming the tokens files directly.

One q I have is I see we're currently flipping bg colors on certain tokens that have the weak/weaker naming - do we also want to do that here or no? I'm thinking against it (and keeping this inverse background only on inverse tokens) but would want your thoughts on this too since we're currently doing this.

Screen Shot 2022-07-19 at 10 06 41 AM

**Review by shleewhite — APPROVED** [review]

File: packages/paste-website/src/components/tokens-list/token-card/token-example/BoxExample.tsx [hunk 6]
```diff
@@ -19,6 +23,7 @@ export const BoxExample: React.FC<BoxExampleProps> = ({
     <Box
       aria-hidden="true"
       backgroundColor={backgroundColor}
+      style={{backgroundColor: highlightColor}}
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/token-card/token-example/BoxExample.tsx:26]: Question: can this be a regular style prop? Noticed a few usages of inline styles throughout the PR and wondering why we had to use those

File: packages/paste-website/src/components/tokens-list/token-card/token-example/TextExample.tsx [hunk 7]
```diff
@@ -21,6 +25,7 @@ export const TextExample: React.FC<TextExampleProps> = ({fontFamily, fontSize, f
         fontSize={fontSize || 'fontSize60'}
         fontWeight={fontWeight}
         lineHeight={lineHeight}
+        style={{color: textColor}}
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/token-card/token-example/TextExample.tsx:28]: Question: im a little unclear about how this interacts with the following line

**Review by TheSisb — APPROVED** [review]

**Pull request merged**